src/graph.rs

```diff
@@ -457,14 +487,75 @@ impl Value {
         }
     }
 
+    pub fn as_list(&self, graph: &Graph) -> Result<&Vec<Value>, ExecutionError> {
+        match self {
+            Value::List(values) => Ok(values),
+            _ => Err(ExecutionError::ExpectedList(format!(
+                "got {}",
+                self.display_with(graph)
+            ))),
+        }
+    }
+
+    /// Coerces this value into a graph node reference, returning an error if it's some other type
+    /// of value.
+    pub fn into_graph_node_ref<'a, 'tree>(
+        self,
+        graph: &'a Graph<'tree>,
+    ) -> Result<GraphNodeRef, ExecutionError> {
+        match self {
+            Value::GraphNode(node) => Ok(node),
+            _ => Err(ExecutionError::ExpectedGraphNode(format!(
+                "got {}",
+                self.display_with(graph)
+            ))),
+        }
+    }
+
+    pub fn as_graph_node_ref<'a, 'tree>(
+        &self,
+        graph: &'a Graph<'tree>,
+    ) -> Result<&GraphNodeRef, ExecutionError> {
```

Comment:
Same as above for `as_{graph,syntax}_node_ref` — `{Graph,Syntax}NodeRef` are just `u32`s under the covers, so I think we should return them by value not by reference